contributing: Update signed commit guidelines (#305)
canonical/sphinx-docs-starter-pack: sp-files/contributing-myst.md

<!-- TODO: Update with your suggestions or drop if excessive -->

To improve contribution tracking, we use the developer certificate of origin
([DCO 1.1](https://developercertificate.org/) and require signed commits (using
the `-S` or `--sign-off` option) for all changes that go into the ACME project.

```{code-block} none
git commit -S -m "acme/component: updated life cycle diagram"
```

Signed commits will have a GPG, SSH, or S/MIME signature that is
cryptographically verifiable, and will be marked with a "Verified" or
"Partially verified" badge in GitHub. This sign off verifies that you made the 
changes or have the right to commit it as an open-source contribution.

To set up locally signed commits and tags, see [GitHub Docs - About commit
signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification).

````{tip}
Configure your Git client to sign commits by default for any local repository by

> running:

Author: Hey @akcano , I didn't want to make this sound like a directive (e.g. "Set up..."). It is more of a 'suggestion' as this may not be needed for everyone for all repos. But probably it can be worded better.

```{code-block} none
git config --global commit.gpgsign true
```

See [GitHub Docs - Signing commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
````

If you made an unsigned commit and encounter the "Commits must have verified
signatures" error when pushing your changes to remote:

> 1. Amend the most recent commit, add a signature without changing the commit

Author: Hey @akcano , off meaning?

For MD you can use "1. " for numbered lists instead of "1." "2." so it's easier to edit and manage.

Same like "1." followed by "#." for RST files.

If that is what you meant.

the `-S` or `---gpg-sign` option) for all changes that go into the ACME project.

```{code-block} none
git commit -S -m "acme/component: updated life cycle diagram"
```

Signed commits will have a GPG, SSH, or S/MIME signature that is
cryptographically verifiable, and will be marked with a "Verified" or
"Partially verified" badge in GitHub. This verifies that you made the changes or
have the right to commit it as an open-source contribution.

To set up locally signed commits and tags, see [GitHub Docs - About commit
signature verification](https://docs.github.com/en/authentication/managing-commit-signature-verification/about-commit-signature-verification).

````{tip}
You can configure your Git client to sign commits by default for any local
repository by running:

```{code-block} none
git config --global commit.gpgsign true
```

See [GitHub Docs - Signing commits](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) for more information.
````

If you've made an unsigned commit and encounter the "Commits must have verified
signatures" error when pushing your changes to the remote:

> 1. Amend the most recent commit, add a signature without changing the commit

Reviewer: ```suggestion
1. Amend the most recent commit by signing it without changing the commit
```

Reviewer: I suggest this rewording because "add a signature" could apply to both signing off and signing.
